InscryptionAPI/Encounters/OpponentManager.cs

```diff
@@ -199,52 +197,58 @@ public static void ProcessBossType(NodeData nodeData)
             BossBattleNodeData bossBattleNodeData = (BossBattleNodeData)nodeData;
 
             // Parse data
-            string value = ModdedSaveManager.RunState.GetValue(InscryptionAPIPlugin.ModGUID, "PreviousBosses");
-            List<int> previousBosses= new List<int>(); // 2,0,1
+            List<Opponent.Type> bosses = RunStateOpponents;
+            if(bosses.Count == 0)
+            {
+                bosses.Add(Opponent.Type.ProspectorBoss);
+                bosses.Add(Opponent.Type.AnglerBoss);
+                bosses.Add(Opponent.Type.TrapperTraderBoss);
+            }
+            
+            bossBattleNodeData.bossType = bosses[RunState.CurrentRegionTier];
+        }
+    }
+
+    public static List<Opponent.Type> RunStateOpponents
+    {
+        get
+        {
+            List<Opponent.Type> previousBosses = new List<Opponent.Type>();
+
+            string value = ModdedSaveManager.RunState.GetValue(InscryptionAPIPlugin.ModGUID, "PreviousBosses"); // 2,0,1
             if (value == null)
```

Comment:
Instead of an empty if-statement handling null values, couldn't you wrap this in an if (value != null)?

Reply:
In this condition I'm checking 
- if its null
- if it contains 
- when it's not null and doesn't contain ,

If if were to do a != null then it's need to nest an if statement which i typically avoid if possible